client/app/pages/queries/query.html

```diff
@@ -102,7 +102,8 @@
               <span class="text-muted">Data Source</span>
               <select ng-disabled="!isQueryOwner" ng-model="query.data_source_id" ng-change="updateDataSource()"
                       ng-options="ds.id as ds.name for ds in dataSources"></select>
-              <a ng-if="dataSource.options.doc_url" ng-href={{dataSource.options.doc_url}}>{{dataSource.type_name}} documentation</a>
+              <a ng-if="dataSource.options.doc_url != ''" ng-href={{dataSource.options.doc_url}}>{{dataSource.type_name}} documentation</a>
+              <span ng-if="dataSource.options.doc_url ==''">{{dataSource.type_name}}</span>
```

Comment:
The purpose of this change is to show the sql dialect regardless of if the doc_url parameter is provided in the data source configuration. Before, it would only show the dialect if there was a doc_url defined. Now, if a doc_url is defined it's a link to the docs with the dialect name, if no doc_url then just print the dialect. 

Does that answer all of your questions/issues except for the missing space?